### DISCUSSION TITLE
reuse existed code

### DISCUSSION THREAD

**Discussion by Freed-Wu** [foxfriends/syncat#17] [Ideas]
Can this project reuse current existed project's code to fasten its development? For example:

<https://github.com/neovim/neovim/blob/607606e3bb7a435ac5a0eba2eb4abde8eff774bb/runtime/lua/vim/filetype.lua#L175-L2031> define the map from file extensions to vim filetype.

<https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/parsers.lua#L29-L48>
define the map from vim filetype to tree-sitter parser name

<https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/parsers.lua#L79-L2285> collect all parsers's information, which can be exported to a [json](https://github.com/grantjenks/py-tree-sitter-languages/pull/53/files#diff-0ab2f44bde72c543a5b3b22415f538059e317e3614cfbea99b7ffe6e13e3c06b) by a [lua script](https://github.com/grantjenks/py-tree-sitter-languages/pull/53/files#diff-46c652b4be9ee78527123a575a38b9d4673a17975990e2fdb23cdac6ba427257)

<https://github.com/nvim-treesitter/nvim-treesitter/tree/master/queries> collect different parsers' captures to map different AST's node type to some unified scope name: [tree-sitter standard names](https://github.com/tree-sitter/tree-sitter/blob/3f44b896852eb7daaa6df4fb778c9bb52c70c815/highlight/src/lib.rs#L20-L72) and [helix extended names](https://docs.helix-editor.com/master/themes.html#syntax-highlighting).

<https://github.com/sharkdp/bat/tree/master/assets/themes> collect some tmThemes which map scope names to color name.

In neovim, it will map scope name to vim highlight group name, then use [vim colorscheme](https://github.com/flazz/vim-colorschemes) to map highlight group to color name, which is a little complex than tmTheme.

**Comment by foxfriends** [comment]
If you wanted to do that to generate some stylesheets and the `languages.toml` automatically, sounds like a great plan. If you do it, feel free to make a PR to add it back here. I won't be doing it though, I don't have the time to maintain that much code right now.

I wouldn't replace the existing stylesheets with this though, the purpose of Syncat was to have CSS-like stylesheets that are more flexible and easy to write/read than I find XML or vimscript or other common highlighter formats to be. If you want to rebuild nvim's highlighter as a standalone CLI, that sounds like a different project than this.
